src/types.ts

```diff
@@ -1893,18 +1893,179 @@ export class ZodUnion<T extends ZodUnionOptions> extends ZodType<
     return this._def.options;
   }
 
-  static create = <
-    T extends Readonly<[ZodTypeAny, ZodTypeAny, ...ZodTypeAny[]]>
+  static create<
```

Comment:
I personally find overloading like this to be a bit of a headache when trying to understand, maintain, and debug things. Additionally, I think it's a bit of a big change to introduce a `ZodType` constructor that actually might return one of two actual `ZodType`s.

Unless @colinhacks or the other maintainers disagree with me on this, I think we can leave out the aliasing for now and (maybe) introduce it later. At the very least, you could cherrypick this change into a separate PR and we can discuss the merits of it without holding up the overall progress on `discriminatedUnion`. Does that seem reasonable?

Reply:
removed